Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeScrollingNodeDelegateIOS.mm

```diff
@@ -223,8 +223,8 @@ - (void)cancelPointersForGestureRecognizer:(UIGestureRecognizer *)gestureRecogni
 
 void ScrollingTreeScrollingNodeDelegateIOS::updateScrollViewForOverscrollBehavior(UIScrollView *scrollView, const WebCore::OverscrollBehavior horizontalOverscrollBehavior, WebCore::OverscrollBehavior verticalOverscrollBehavior, AllowOverscrollToPreventScrollPropagation allowPropogation)
 {
-    scrollView.bouncesHorizontally = horizontalOverscrollBehavior != OverscrollBehavior::None;
-    scrollView.bouncesVertically = verticalOverscrollBehavior != OverscrollBehavior::None;
+    scrollView.bouncesHorizontally &= horizontalOverscrollBehavior != OverscrollBehavior::None;
+    scrollView.bouncesVertically &= verticalOverscrollBehavior != OverscrollBehavior::None;
```

Comment:
This isn't the right way to fix the bug.

It needs to be fixed such that a webkit client can read and write the value of `scrollView.bounces` and get consistent behavior (i.e. they get back the value they set). This needs to work even when webkit internally sets the value because of OverscrollBehavior.

The way we do this is to put code in WKScrollView to track the internal state; see `_setScrollEnabledInternal` etc.

There will be API tests that verify this behavior, and you should make one for this case too.